packages/sync-actions/src/zones-actions.js

```diff
@@ -25,20 +25,45 @@ export function actionsMapLocations(diff, oldObj, newObj) {
       action: 'addLocation',
       location: newLocation,
     }),
-    [REMOVE_ACTIONS]: oldLocation => ({
-      action: 'removeLocation',
-      location: oldLocation,
-    }),
-    [CHANGE_ACTIONS]: (oldLocation, newLocation) => [
-      {
-        action: 'removeLocation',
-        location: oldLocation,
-      },
-      {
-        action: 'addLocation',
-        location: newLocation,
-      },
-    ],
+    [REMOVE_ACTIONS]: oldLocation =>
+      // We only add the action if the location is not included in the new object.
+      !newObj.locations.find(
+        location => location.country === oldLocation.country
+      )
+        ? {
+            action: 'removeLocation',
+            location: oldLocation,
+          }
+        : null,
+    [CHANGE_ACTIONS]: (oldLocation, newLocation) => {
+      const result = []
+
+      // We only remove the location in case that the oldLocation is not
+      // included in the new object
+      if (
+        !newObj.locations.find(
```

Comment:
Same.

Reply:
even go as far as to put this logic into a util